File: python/a27.py
# A27 Calculate GCD
# submission: https://atcoder.jp/contests/tessoku-book/submissions/38566598

def gcd(x: int, y: int) -> int:
    if x % y == 0:
        return y
    return gcd(y, x % y)

a, b = map(int, input().split())

print(gcd(a, b))

File: rust/tessoku-book/src/bin/a27.rs
// A27 Calculate GCD
// submission: https://atcoder.jp/contests/tessoku-book/submissions/38566660

use proconio::input;

fn gcd(x: u32, y: u32) -> u32 {
    if x % y == 0 {
        y
    } else {
        gcd(y, x % y)
    }
}

fn main() {
    input!{
        a: u32,
        b: u32,
    }
    println!("{}", gcd(a, b));
}
